kms/api/Kms.Samples.Tests/CreateKeyAsymmetricDecryptTests.cs

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Google.Cloud.Kms.V1;
+using Xunit;
+
+[Collection(nameof(KmsFixture))]
+public class CreateKeyAsymmetricDecryptTest
+{
+    private readonly KmsFixture _fixture;
+    private readonly CreateKeyAsymmetricDecryptSample _sample;
+
+    public CreateKeyAsymmetricDecryptTest(KmsFixture fixture)
+    {
+        _fixture = fixture;
+        _sample = new CreateKeyAsymmetricDecryptSample();
+    }
+
+    [Fact]
+    public void CreatesKey()
+    {
+        // Run the sample code.
+        var result = _sample.CreateKeyAsymmetricDecrypt(
+            projectId: _fixture.ProjectId, locationId: _fixture.LocationId, keyRingId: _fixture.KeyRingId,
+            id: _fixture.RandomId());
+
+        // Get the key.
+        KeyManagementServiceClient client = KeyManagementServiceClient.Create();
+        var key = client.GetCryptoKey(new GetCryptoKeyRequest
```

Comment:
Why are we avoiding the convenience overloads? Is this now a deprecated feature of GAPIC?

I would write this as
`var key = client.GetCryptoKey(result.CryptoKeyName);`

(Question applies to all of these tests and samples)

Reply:
If the protos are correctly annotated then we should be generating these overloads. I agree that if they are there, it's better to always use the simplest one.

Reply:
They exist, but I intentionally avoided them. The other statically-typed languages (Go and Java) both prefer the "construct and accept the entire request object" pattern, and I tried to be consistent here. I've also heard feedback that the magical methods that accept arguments often aren't flexible enough for advanced use cases, and customers have to drop down to this way anyway.

It's not a hill upon which I stake my claim, but it feels good to be consistent?

Reply:
I do like the consistency aspect of using the request object. But I do think that when the overloads are enough they should be used for simplicity. If there's a complex use case that is not covered by the overloads then I would suggest writing a sample that covers that complex use case and shows how to create and populate the request object accordingly.

Reply:
Ack, updated to use overloads.